Core/include/Acts/Propagator/MultiEigenStepperLoop.ipp

```diff
@@ -0,0 +1,200 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+namespace Acts {
+
+template <typename E, typename R, typename A>
+auto MultiEigenStepperLoop<E, R, A>::boundState(State& state,
+                                                const Surface& surface,
+                                                bool transportCov) const
+    -> Result<BoundState> {
+  if (numberComponents(state) == 1) {
+    return SingleStepper::boundState(state.components.front().state, surface,
+                                     transportCov);
+  } else {  // Do the combination
+    SmallVector<std::pair<double, BoundTrackParameters>> states;
+    double accumulatedPathLength = 0.0;
+    int failedBoundTransforms = 0;
+
+    for (auto i = 0ul; i < numberComponents(state); ++i) {
+      auto bs = SingleStepper::boundState(state.components[i].state, surface,
+                                          transportCov);
+
+      if (bs.ok()) {
+        states.push_back(
+            {state.components[i].weight, std::get<BoundTrackParameters>(*bs)});
+        accumulatedPathLength +=
```

Comment:
This implicitly implies that the navigation path is a weighted mean of all paths, will this hold for all merging policies? 
(probably)

I guess the accumulatedPathLength is anyway only needed/used for the path length aborter and the path limit settings in this scope, so I *guess* this should be fine.

Reply:
Yeah, in all places where a single `pathLength`-representation for all components is needed, I did a weighted sum (e.g. for the return value of the `MultiEigenStepperLoop::step(...)`-function.
The pathlength of such a multi-component-propagation isn't really defined I think, and this is the value which comes closest to it. 
One could also use the pathlenght of the maximum-weight component e.g., but in normal use cases this should not make a huge difference.